### PR TITLE
Harden CSV validation

### PR DESCRIPTION
**Pull request by brianhelba**
See individual commits.

### CODE REVIEW

**Comment by brianhelba** [comment]
@danlamanna Previously, by opening in binary mode, we were not doing automatic newline conversion. This change makes the validator potentially more tolerant of different newline conventions. Do you have any concerns that allowing more diversity in accepted CSVs would make them more difficult to work with later in research settings?

File: isic_challenge_scoring/load_csv.py [hunk 1]
```diff
@@ -27,11 +27,19 @@ def parse_truth_csv(csv_file_stream: TextIO) -> Tuple[pd.DataFrame, pd.DataFrame
 
 
 def parse_csv(csv_file_stream: TextIO, categories: pd.Index) -> pd.DataFrame:
+    if csv_file_stream.read(2000).count('\n') < 2:
+        # Heuristic: if there aren't 2 newlines in the first 2000 characters, it's probably invalid,
```
Comment (brianhelba) [inline isic_challenge_scoring/load_csv.py:31]: Reasonable?

File: isic_challenge_scoring/load_csv.py [hunk 2]
```diff
@@ -27,11 +27,19 @@ def parse_truth_csv(csv_file_stream: TextIO) -> Tuple[pd.DataFrame, pd.DataFrame
...
     try:
         probabilities = pd.read_csv(csv_file_stream, header=0, index_col=False)
-    except pd.errors.ParserError as e:
-        # TODO: Test this case
-        raise ScoreException(f'Could not parse CSV: "{str(e)}"')
+    except (pd.errors.ParserError, pd.errors.EmptyDataError) as e:
+        # TODO: Test something that generates a ParserError
+        raise ScoreException(f'Could not parse CSV: "{str(e)}".')
+    except IndexError:
+        raise ScoreException('Could not parse CSV: mismatched header row.')
```
Comment (danlamanna) [inline isic_challenge_scoring/load_csv.py]: Can this be a bit more human readable?

File: isic_challenge_scoring/load_csv.py (hunk 1 above)
Comment (danlamanna) [reply isic_challenge_scoring/load_csv.py:31]: Yeah

File: isic_challenge_scoring/load_csv.py (hunk 2 above)
Comment (brianhelba) [reply isic_challenge_scoring/load_csv.py]: Do you have a suggestion?

Comment (danlamanna) [reply isic_challenge_scoring/load_csv.py]: "Could not parse CSV: differing number of columns" ?

Comment (brianhelba) [reply isic_challenge_scoring/load_csv.py]: Better now?

**Review by danlamanna — APPROVED** [review]

**Pull request merged**